test/aqua.jl

```diff
@@ -0,0 +1,36 @@
+using Test
+using StatsBase
+using Aqua
+
+@testset "Aqua tests (performance)" begin
+    # This tests that we don't accidentally run into
+    # https://github.com/JuliaLang/julia/issues/29393
+    # Aqua.test_unbound_args(StatsBase)
+    ua = Aqua.detect_unbound_args_recursively(StatsBase)
+    @test length(ua) == 0
+
+    # See: https://github.com/SciML/OrdinaryDiffEq.jl/issues/1750
+    # Test that we're not introducing method ambiguities across deps
+    ambs = Aqua.detect_ambiguities(StatsBase; recursive = true)
+    # Uncomment for debugging:
+    # for method_ambiguity in ambs
+    #     @show method_ambiguity
+    # end
+    @test length(ambs) ≤ 5
```

Comment:
Is there a simple way to check which functions are ambiguous?

Reply:
We can use `Aqua.test_ambiguities(StatsBase; recursive = true, exclude = [StatsBase.TestStat, Base.:(==)])`. However, this doesn't test the number of ambiguities, only the functions.. Testing the number of ambiguities is nice, IMO, because we can even write this as:
```julia
    @test length(ambs) ≤ 5
    @test_broken length(ambs) < 5
```
so that we're alerted that it's fixed. It's also just easier to spread across the ecosystem than to collect all the functions after printing them all out..

Reply:
I'm happy to change it if it means getting the PR merged.